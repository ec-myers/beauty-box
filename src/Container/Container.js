import React from 'react';
import { PropTypes } from 'prop-types';
import Product from '../Product/Product';
import './Container.scss';

const Container = ({productType, toggleCollection, type, collection , allCategories}) => {
  if (type === 'collection') {
    let collectionList = collection.map(product => <Product key={product.id} product={product} toggleCollection={toggleCollection} /> );

    return (
      <main className='Container'>
        {collectionList}
      </main>
    )
  } else if (type === 'shopall') {
    let allCategoriesList = allCategories.map(product => <Product key={product.id} product={product} toggleCollection={toggleCollection} />);

    return (
      <main className='Container'>
        {allCategoriesList}
      </main>
    )
  } else {
    const productList = productType.map(product => <Product key={product.id}product={product} toggleCollection={toggleCollection} />);

    return (
      <main className = 'Container'>
        {productList}
      </main>
    )
  }
}

export default Container;

Container.propTypes = {
  productType: PropTypes.array,
  toggleCollection: Proptypes.func,
  type: PropTypes.string,
  collection: PropTypes.array,
  allCategories: PropTypes.array
}