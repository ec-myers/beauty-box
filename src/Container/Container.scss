.Container {
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(4, 1fr);
  height: 100vh;
  overflow: scroll;
  padding: 35px;
}