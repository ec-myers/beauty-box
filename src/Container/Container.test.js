import React from 'react';
import { shallow } from 'enzyme';
import Container from './Container';

describe('Container', () => {
  it('should match the snapshot if there is a productType', () => {
    let toggleCollection = jest.fn();
    let type = '';
    let collection = [{
      id: 1047,
      brand: "colourpop",
      name: "Blotted Lip",
      image_link: "https://file.com",
      product_link: "https://colourpop.com/",
      description: "Description",
      tag_list: ['tag1', 'tag2']
    }];
    let allCategories = [{
      id: 1047,
      brand: "colourpop",
      name: "Blotted Lip",
      image_link: "https://file.com",
      product_link: "https://colourpop.com/",
      description: "Description",
      tag_list: ['tag1', 'tag2']
    }];
    let productType = ['mascara'];

    const collectionWrapper = shallow(
      <Container
        productType={productType}
        toggleCollection={toggleCollection}
        type={type}
        collection={collection}
        allCategories={allCategories}
        />
    );

    expect(collectionWrapper).toMatchSnapshot();
  });

  it('should match the snapshot if type is collection', () => {
    let toggleCollection = jest.fn();
    let type = 'collection';
    let collection = [{
      id: 1047,
      brand: "colourpop",
      name: "Blotted Lip",
      image_link: "https://file.com",
      product_link: "https://colourpop.com/",
      description: "Description",
      tag_list: ['tag1', 'tag2']
    }];
    let allCategories = [{
      id: 1047,
      brand: "colourpop",
      name: "Blotted Lip",
      image_link: "https://file.com",
      product_link: "https://colourpop.com/",
      description: "Description",
      tag_list: ['tag1', 'tag2']
    }];
    let productType = [''];

    const collectionWrapper = shallow(
      <Container
        productType={productType}
        toggleCollection={toggleCollection}
        type={type}
        collection={collection}
        allCategories={allCategories}
      />
    );

    expect(collectionWrapper).toMatchSnapshot();
  });

  it('should match the snapshot if type is shopall', () => {
    let toggleCollection = jest.fn();
    let type = 'shopall';
    let collection = [{
      id: 1047,
      brand: "colourpop",
      name: "Blotted Lip",
      image_link: "https://file.com",
      product_link: "https://colourpop.com/",
      description: "Description",
      tag_list: ['tag1', 'tag2']
    }];
    let allCategories = [{
      id: 1047,
      brand: "colourpop",
      name: "Blotted Lip",
      image_link: "https://file.com",
      product_link: "https://colourpop.com/",
      description: "Description",
      tag_list: ['tag1', 'tag2']
    }];
    let productType = [''];

    const collectionWrapper = shallow(
      <Container
        productType={productType}
        toggleCollection={toggleCollection}
        type={type}
        collection={collection}
        allCategories={allCategories}
      />
    );

    expect(collectionWrapper).toMatchSnapshot();
  });
});