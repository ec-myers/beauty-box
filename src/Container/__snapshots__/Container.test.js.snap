// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot if there is a productType 1`] = `
<main
  className="Container"
>
  <Connect(Product)
    product="mascara"
    toggleCollection={[MockFunction]}
  />
</main>
`;

exports[`Container should match the snapshot if type is collection 1`] = `
<main
  className="Container"
>
  <Connect(Product)
    key="1047"
    product={
      Object {
        "brand": "colourpop",
        "description": "Description",
        "id": 1047,
        "image_link": "https://file.com",
        "name": "Blotted Lip",
        "product_link": "https://colourpop.com/",
        "tag_list": Array [
          "tag1",
          "tag2",
        ],
      }
    }
    toggleCollection={[MockFunction]}
  />
</main>
`;

exports[`Container should match the snapshot if type is shopall 1`] = `
<main
  className="Container"
>
  <Connect(Product)
    key="1047"
    product={
      Object {
        "brand": "colourpop",
        "description": "Description",
        "id": 1047,
        "image_link": "https://file.com",
        "name": "Blotted Lip",
        "product_link": "https://colourpop.com/",
        "tag_list": Array [
          "tag1",
          "tag2",
        ],
      }
    }
    toggleCollection={[MockFunction]}
  />
</main>
`;
